geolink_formatter/parser.py

```diff
@@ -98,14 +101,26 @@ def from_string(self, xml):
 
         for document_el in root.iter('document'):
             doc_id = document_el.attrib.get('id')
+            doctype = document_el.attrib.get('doctype')
+
+            # Mangle doc_id for notices. While IDs are unique between decrees
+            # and edicts, this is not the case when adding notices to the mix.
+            if doctype == 'notice':
+                doc_id += doctype
+
             if doc_id and doc_id not in [doc.id for doc in documents]:
                 files = list()
                 for file_el in document_el.iter('file'):
                     href = file_el.attrib.get('href')
                     if self.host_url and not href.startswith(u'http://') and not href.startswith(u'https://'):
                         href = u'{host}{href}'.format(host=self.host_url, href=href)
+
+                    # NOTE: The XML attribute 'title' contains the filename in
+                    # truth, and the attribute 'description' is the true title
+                    # of the file (may be empty).
                     files.append(File(
-                        title=file_el.attrib.get('title'),
+                        title=file_el.attrib.get('description'),
```

Comment:
This change might break backward compatibility for users currently using the attribute `title` as file name. I would like to keep things simple and use the attributes from the geoLink schema (`title` and `description`).